typed/typed.go

```diff
@@ -59,13 +59,35 @@ func (tv TypedValue) Validate() error {
 	return errs
 }
 
+// ToFieldSet creates a set containing every leaf field mentioned in tv, plus
+// any validation errors. A (perhaps partial) field set is returned even if
+// there are validation errors.
```

Comment:
> A (perhaps partial) field set is returned even if there are validation errors.

Why?

Reply:
To let us roll this out gradually even if there's old objects that don't validate against the schema. Open to better ways to do this.

I guess we could require objects to validate against the schema if you're using the new apply?

Reply:
That sounds reasonable, yes. Also, I suspect people who use apply already go through client-side validation (you can skip it, but it's enabled by default).